lms/djangoapps/instructor/views/api.py

```diff
@@ -1024,9 +1095,9 @@ def get_problem_responses(request, course_id):
     # The name of the POST parameter is `problem_location` (not pluralised) in
     # order to preserve backwards compatibility with existing third-party
     # scripts.
-    problem_locations = request.POST.get('problem_location', '')
+    problem_locations = getattr(request, 'data', request.POST).get('problem_location', '')
```

Comment:
If we're coming to this method from the DRF view above, this data will be available in `request.data`, while for the Django view it will be in `request.POST`.

Reply:
I've split the parsing of this part out separately so that the new API can use a better format.